keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -324,47 +335,109 @@ def next(prompt, state, index):
 
     def generate(
         self,
-        prompt,
-        max_length,
+        inputs,
+        max_length=None,
+        batch_size=None,
+        return_tensors=False,
     ):
-        """Generate text.
+        """Generate text given prompt `inputs`.
+
+        This method generates text based on given `prompt`.The sampling used for
+        generation can be set in the `compile` method.
 
-        This method generates text based on given `prompt`. Generation will
-        continue until `max_length` is met, and all tokens generated after
-        `end_token` will be truncated. The sampling strategy can be set in
-        the `compile` method.
+        If `inputs` are a batched `tf.data.Dataset` or `batch_size` is provided,
+        outputs will be generated "batch-by-batch" and concatenated. Otherwise,
+        all inputs will be handled in a single generation pass.
+
+        If a `preprocessor` is attached to the model, `inputs` should be
```

Comment:
You need to discuss `return_tensors` here, depending on the outcome of my questions below

Reply:
See above responses!

Reply:
Removed return tensors for a follow up PR.